src/plugins/data_source/server/routes/data_source_connection_validator.ts

```diff
@@ -35,4 +35,27 @@ export class DataSourceConnectionValidator {
       throw createDataSourceError(e);
     }
   }
+
+  async fetchDataSourceVersion() {
```

Comment:
Should this be called fetchDataSourceMetadata or do we want to create another API to fetch enabled plugins?

Reply:
yes, this is just the first step to fetch the version number, next one is to fetch the installed plugins which would be consolidated into one method say fetchDataSourceMetadata with multiple api calls behind the scene